File: 1221-split-a-string-in-balanced-strings/1221-split-a-string-in-balanced-strings.java
class Solution {
    public int balancedStringSplit(String s) {
        int cnt = 0, res = 0;
        for(int i = 0 ; i < s.length() ; i++){
            if(s.charAt(i) == 'L'){
                cnt++;
            }else{
                cnt--;
            }
            if(cnt == 0){
                res++;
            }
        }
        return res;
    }
}

File: 1221-split-a-string-in-balanced-strings/1221-split-a-string-in-balanced-strings.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        cnt = 0
        res = 0
        for ch in s:
            if ch == 'L':
                cnt += 1
            else:
                cnt -= 1
            if cnt == 0:
                res += 1
                
        return res